app/models/reports/repository.rb

```diff
@@ -18,49 +20,115 @@ def initialize(regions, periods:, with_exclusions: false)
 
     delegate :cache, :logger, to: Rails
 
+    def for_region_and_period(region, period)
+      raise ArgumentError, "Repository does not include region #{region.slug}" unless regions.include?(region)
+      raise ArgumentError, "Repository does not include period #{period}" unless periods.include?(period)
+      RegionAndPeriodFetcher.new(self, region, period)
+    end
+
+    # Returns assigned patients for a region. NOTE: We grab and cache ALL the counts for a particular region with one SQL query
+    # because it is easier and fast enough to do so. We still return _just_ the periods the Repository was created with
+    # to conform to the same interface as all the other queries here.
+
+    # Returns a Hash in the shape of:
+    # {
+    #    region_slug: { period: value, period: value },
+    #    region_slug: { period: value, period: value }
+    # }
+    def assigned_patients_count
+      full_assigned_patients_counts.each_with_object({}) do |(entry, result), results|
+        values = periods.each_with_object({}) { |period, region_result| region_result[period] = result[period] }
+        results[entry.region.slug] = values
+      end
+    end
+
+    def full_assigned_patients_counts
```

Comment:
Should these methods be memoized? Pulling from the cache may be very efficient, but I would guess there's still some inefficiency here.